libcore/src/events/TrainEvent.cpp

```diff
@@ -0,0 +1,320 @@
+#include "TrainEvent.h"
+
+#include "geometry/SubRoom.h"
+#include "geometry/Wall.h"
+
+void TrainEvent::Process()
+{
+    LOG_INFO("{}", GetDescription());
+    switch(_action) {
+        case EventAction::TRAIN_ARRIVAL:
+            TrainArrival();
+            break;
+        case EventAction::TRAIN_DEPARTURE:
+            TrainDeparture();
+            break;
+        default:
+            throw std::runtime_error("Wrong EventAction for TrainEvent!");
+    }
+}
+
+std::string TrainEvent::GetDescription() const
```

Comment:
I would like to have this as template function having the EventAction as Non-Type Template parameter.

Reply:
see #764